### DISCUSSION TITLE
how to use  VC100 for MSBuild ?

### DISCUSSION THREAD

**Discussion by Kerry-Cho** [community/community#25438] [Actions]
<p>I want to build a project developed with Visual studio 2010. The installed version is 2017 or 2019 and I want to know how to build another version.</p>
<p>Thanks.</p>

**Comment by wipe2238** [comment, marked as answer]
<p>Install VS2010.</p>
<p>I have a project for VS2010 which i cannot update (yet) without breaking it, and that’s the only <em>working</em> solution i found. While on local you can use VS2017 to work with files, you still need VS2010 installed for v100 toolset…</p>
<p>Anyway, someone put <a href="https://chocolatey.org/packages/VCExpress2010" rel="nofollow noopener">all what’s needed</a> on Chocolatey, so under Actions it’s one-liner…</p>
<pre><code>- run: choco install vcexpress2010
</code></pre>
<p>…which <a href="https://gist.github.com/choco-bot/32f2c2daaf14a5ae24d86bb5f4b537b2#file-install-txt-L354" rel="nofollow noopener">downloads</a> and <a href="https://gist.github.com/choco-bot/32f2c2daaf14a5ae24d86bb5f4b537b2#file-install-txt-L370" rel="nofollow noopener">runs</a></p>
<pre><code>https://download.microsoft.com/download/1/D/9/1D9A6C0E-FC89-43EE-9658-B9F0E3A76983/vc_web.exe
</code></pre>
<p>This adds ~5-6 minutes to job run time, just to grab and prepare VS2010; not so great, but not so terrible either. Would love to see better/faster solution for this one, but currently it’s all i got.</p>

**Comment by Kerry-Cho** [comment]
<p>Oh. Thank a lot ^^</p>

**Comment by Kerry-Cho** [comment]
<p>I don’t know how to run vcexpress after installation</p>
<p>Also Can I patch the sdk separately?</p>
<p>just, I would like to use like below code</p>
<p>start msbuild env</p>
<pre><code>call "C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\vcvarsall.bat" x86
</code></pre>
<p>and build</p>
<pre><code>msbuild Build.proj /t:%TARGET% /p:Platform="%PLATFORM%";PlatformToolset="%TOOLSET%"
</code></pre>
<p>Thanks</p>

**Comment by wipe2238** [comment]
<blockquote>
<p><a class="mention" href="/kerry-cho">@kerry-cho</a> wrote:</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>vcvars</p>
</blockquote>
<p>Sweet gods. I switched to CMake so long ago i totally forgot it exists…</p>
<p>If you can (and i’m fully aware it might be overkill for some projects), i’d recommend doing the same, so you don’t have to worry about such stuff anymore:</p>
<pre><code>md Build
cd Build
cmake -G "Visual Studio 10 2010" ..
cmake --build . --config Release
</code></pre>
<p>That’s more or less what i’m using, leaving all dirty work to smart people maintaining CMake.</p>
<hr>
<p>If that’s not an option… oh boy.</p>
<p>In theory this should work…</p>
<pre><code>- run: "C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\vcvarsall.bat" x86
- run: "C:\Path\To\MSBuild.exe" ...
</code></pre>
<p>…but gods knows if Actions will keep env variables set inside <em>.bat</em> or not, didn’t tried yet. If they doesn’t for w/e reason, that’s not a [big] problem; you can extract all “set” from vcvars into something like this…</p>
<pre><code>@set PATH=C:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE;%PATH%
@set PATH=C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN;%PATH%
@set LIB=C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\LIB;%LIB%
@set LIB=C:\Program Files (x86)\Microsoft SDKs\Windows\v6.0A\Lib;%LIB%
@set LIB=C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\Lib;%LIB%
@set INCLUDE=C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include;%INCLUDE%
</code></pre>
<p>…and convert it to fancy Actions step:</p>
<pre><code>- name: Environment
run: |
         echo ::add-path::C:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE
         echo ::add-path::C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN
         echo ::set-env name=LIB::C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\LIB;%LIB%
         echo ::set-env name=LIB::C:\Program Files (x86)\Microsoft SDKs\Windows\v6.0A\Lib;%LIB%
         echo ::set-env name=LIB::C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\Lib;%LIB%
         echo ::set-env name=INCLUDE::C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include;%INCLUDE%
shell: cmd
</code></pre>
<p>That should configure environment just like vcvars would, for all later steps; see <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/development-tools-for-github-actions#add-a-system-path-add-path" rel="nofollow noopener">::add-path</a> and <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/development-tools-for-github-actions#set-an-environment-variable-set-env" rel="nofollow noopener">::set-env</a> docs for details.</p>
<p>What’s left is finding and running MSBuild; i have no bloody idea where installer puts it, neither i have any machine with VS2010 installed currently to check. Explore! :&gt; Once you find it, just add it to PATH and…</p>
<pre><code>- run: msbuild OhNoesLegacyCode.sln [options]
</code></pre>
<p>…like there’s no tommorow.</p>
<p>Easy, right? :slight_smile:</p>
<p>Wish i could help more than just wild guessing above, but it’s been years since i compiled anything on windows from scratch using vcvars + msbuild…</p>

**Comment by Kerry-Cho** [comment]
<p>Thanks you so much , sloved with your help.</p>
<p>The configuration time is a bit longer, but it doesn’t matter because the build time is longer.</p>
<p>Below is a Sample Repo</p>
<p><a href="https://github.com/kerry-Cho/Actions_msbuild2010" rel="nofollow noopener">https://github.com/kerry-Cho/Actions_msbuild2010</a></p>

**Comment by AraHaan** [comment]
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/AraHaan/pthmgr/runs/1279389879?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/15173749/181091135-48033d98-897c-4c0a-85c9-180759623052.png" class="thumbnail onebox-avatar" width="161" height="161">

<h3><a href="https://github.com/AraHaan/pthmgr/runs/1279389879?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">AraHaan/pthmgr</a></h3>

<p>Windows Path Manager. Contribute to AraHaan/pthmgr development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
I tried your config and it does not seem to want to work anymore.
